FirebaseAppDistribution/Sources/FIRAppDistributionMachO.m

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <CommonCrypto/CommonHMAC.h>
+#include <mach-o/arch.h>
+#import <mach-o/fat.h>
+#import <mach-o/loader.h>
+#import "FIRAppDistributionMachO+Private.h"
+#import "FIRAppDistributionMachOSlice+Private.h"
+
+@implementation FIRAppDistributionMachO
+
+- (instancetype)initWithPath:(NSString*)path {
+  self = [super init];
+
+  if (self) {
+    _file = [NSFileHandle fileHandleForReadingAtPath:path];
```

Comment:
Optional: a rule we have adopted in Crashlytics is to always use `_vars` in constructors, and `self.vars` in the rest of the class. The reason is:
1) In constructors, you don't want to set the `self.var` because it can cause other side effects or an infinite loop in some cases if you have defined a setter, `setVar` method for that variable
2) In the rest of the code, you may want to later define a custom getter for `var`, and `self.var` reads better

Reply:
Using `_var` does not call `getVar`, but `self.var` does. By default all properties have an invisible setter and getter, and you can override it by defining your own

Reply:
good call, done